### PR TITLE
[JavaScript] Add Bot Authentication SNI sample bot 

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR adds a new sample for Javascript_nodejs that shows how to authenticate the Azure bot using the Subject Name/Issuer from MSAL.

## Proposed Changes
  - Added the new sample _**bot-authentication-sni**_.
  - Added a simple bot conversation to validate the authentication.
  - Added README file with the instructions and information about the sample and how to create a SSL/TSL certificate.

## Testing
This image shows the new _**bot-authentication-sni**_ sample working.
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,6 +186,7 @@ A [collection of **experimental** samples](./experimental) exist, intended to pr
 [js#81]:samples/javascript_nodejs/81.skills-skilldialog
 [js#82]:samples/javascript_nodejs/82.skills-sso-cloudadapter
 [js#84]:samples/javascript_nodejs/84.bot-authentication-certificate
+[js#85]:samples/javascript_nodejs/84.bot-authentication-sni
```
Comment (ceciliaavila) [inline README.md]: Change this to the right sample number.

File: samples/javascript_nodejs/85.bot-authentication-sni/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+### Using local environment
+
+1.  Run the following command in a local PowerShell
+
+```
+$cert = New-SelfSignedCertificate -CertStoreLocation "Cert:\CurrentUser\My" -Subject "CN=sampleAppCert" -KeySpec KeyExchange
+```
+
+1.  Then, type _Manage User Certificates_ in the Windows search bar and hit enter
+2.  The certificate will be in the _user certificates_ folder, under _personal_.
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/README.md]:
```suggestion
2.  The certificate will be located in the _user certificates_ folder, under _personal_ directory.
```

File: samples/javascript_nodejs/85.bot-authentication-sni/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+## How to create an SSL/TLS certificate
+
+There are two possible options to create SSL/TSL certificate. Below is a step-by-step description of each one:
+
+### Using local environment
+
+1.  Run the following command in a local PowerShell
+
+```
+$cert = New-SelfSignedCertificate -CertStoreLocation "Cert:\CurrentUser\My" -Subject "CN=sampleAppCert" -KeySpec KeyExchange
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/README.md]: Here we should use tags to let the users know which values they need to provide. For example:
```suggestion
$cert = New-SelfSignedCertificate -CertStoreLocation "<directory-to-store-certificate>" -Subject "CN=<certificate-name>" -KeySpec KeyExchange
```

File: samples/javascript_nodejs/85.bot-authentication-sni/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+1.  Then, type _Manage User Certificates_ in the Windows search bar and hit enter
+2.  The certificate will be in the _user certificates_ folder, under _personal_.
+3.  Export the certificate to _pfx_ format including the key(The default location is _system32_ folder).
+4.  Go to the certificate location and run the following command to generate a _pem_ file:
+
+```
+OpenSSL pkcs12 -in sampleAppCert.pfx -out c:\sampleAppCert.pem –nodes
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/README.md]: The same here with the tags.

File: samples/javascript_nodejs/85.bot-authentication-sni/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+5.  Upload the generated certificate to the Azure app registration.
+
+### Using KeyVault
+
+1.  Create a KeyVault resource and assign _the KeyVault Administrator_ role to have permission to create a new certificate.
+2.  Under the Certificates section, hit on Generate/Import, complete the form, and create the certificate in PEM format.
+3.  Go to the details of the certificate that you created and enable it.
+4.  Download the certificate in CER format and then upload it to the Azure app registration.
+
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/README.md:113]: We should add a section indicating which values to set in the bot's .env file.

File: samples/javascript_nodejs/85.bot-authentication-sni/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    "repository": {
+        "type": "git",
+        "url": "https://github.com"
+    },
+    "dependencies": {
+        "@azure/identity": "^3.3.0",
+        "@azure/keyvault-certificates": "^4.7.0",
+        "@azure/keyvault-secrets": "^4.7.0",
+        "botbuilder": "~4.20.0",
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/package.json]: Should we target version 4.21.0? It's not published yet, but version 4.20 doesn't contain the MSAL support this sample needs.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: samples/javascript_nodejs/85.bot-authentication-sni/package.json (hunk 6 above)
Comment (ceciliaavila) [reply samples/javascript_nodejs/85.bot-authentication-sni/package.json]: Or maybe we should use the `^` symbol.

File: samples/javascript_nodejs/85.bot-authentication-sni/README.md [hunk 7]
```diff
@@ -10,10 +10,16 @@ This bot has been created using [Bot Framework](https://dev.botframework.com/),
 
 ```bash
 
+
+
 # determine node version
 
+
+
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/README.md]: I think this space is too much.

File: samples/javascript_nodejs/85.bot-authentication-sni/.env [hunk 8]
```diff
@@ -1,6 +1,4 @@
-MicrosoftAppType=
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/85.bot-authentication-sni/.env]: I think we should keep the AppType at least. This should work for singletenant.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted to 3914

**Pull request closed without merging**